### PR TITLE
Enable BlobId v2 write.

### PR DESCRIPTION
**Pull request by xiahome**
This PR makes all new blobIds to be serialized in V2.

Tets: clean build

Verified from a local deployment:
```
$ curl -i -H "x-ambry-service-id : CUrlUpload"  -H "x-ambry-owner-id : `whoami`" -H "x-ambry-content-type : image/gif" -H "x-ambry-um-description : Demonstration Image" http://localhost:1174/ --data-binary @frontend.log
HTTP/1.1 201 Created
Location: /AAIA____AAAAAQAAAAAAAAAAAAAAJGM5NmE4NDM4LTQzMmItNGUyZS05MWRhLWRlNDZiNWQ4MDA5OA
Date: Tue, 12 Sep 2017 22:46:09 GMT
Content-Length: 0
x-ambry-creation-time: Tue, 12 Sep 2017 22:46:09 GMT

$ curl -i http://localhost:1174/AAIA____AAAAAQAAAAAAAAAAAAAAJGM5ZmRmOGJiLTY1OGQtNDZmNy05YWQ5LTM0ZTcwN2E3NzNkNA/BlobInfo
HTTP/1.1 200 OK
Date: Tue, 12 Sep 2017 22:46:33 GMT
Last-Modified: Tue, 12 Sep 2017 22:44:25 GMT
x-ambry-blob-size: 5416
x-ambry-service-id: CUrlUpload
x-ambry-creation-time: Tue, 12 Sep 2017 22:44:25 GMT
x-ambry-private: false
x-ambry-content-type: image/gif
x-ambry-owner-id: mxia
x-ambry-um-description: Demonstration Image
Content-Length: 0

$ curl http://localhost:1174/AAIA____AAAAAQAAAAAAAAAAAAAAJGM5ZmRmOGJiLTY1OGQtNDZmNy05YWQ5LTM0ZTcwN2E3NzNkNA > /tmp/a.log
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5416  100  5416    0     0   108k      0 --:--:-- --:--:-- --:--:--  110k

$ vi /tmp/a.log

$ curl -i -X DELETE http://localhost:1174/AAIA____AAAAAQAAAAAAAAAAAAAAJGM5ZmRmOGJiLTY1OGQtNDZmNy05YWQ5LTM0ZTcwN2E3NzNkNA
HTTP/1.1 202 Accepted
Date: Tue, 12 Sep 2017 22:47:34 GMT
Content-Length: 0
```

### CODE REVIEW

File: ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java [hunk 1]
```diff
@@ -71,7 +71,7 @@
   static final short BLOB_ID_V1 = 1;
   // version 2 of the serialized format
```
Comment (pnarayanan) [inline ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java]: these comments are unnecessary.

File: ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java [hunk 2]
```diff
@@ -510,58 +510,78 @@ public void injectionAccountAndContainerForPutTest() throws Exception {
...
+   * The {@link AccountService} is prepopulated with a reference account and {@link Account#UNKNOWN_ACCOUNT}. The expected
+   * behavior should be:
+   * <pre>
+   *   AId in blobId    CId in blobId     expected Error      injected account      injected container
+   *    UNKNOWN           UNKNOWN          NotFound            UNKNOWN               UNKNOWN            This can succeed if the blob exists in backend.
+   * </pre>
+   * @throws Exception
+   */
+  @Test
+  public void injectionAccountAndContainerForGetHeadDeleteBlobIdV1Test() throws Exception {
```
Comment (pnarayanan) [inline ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java]: Is this test necessary? Do you need to check anymore how BlobId V1 treats account and container ids passed to it as this path will not be exercised anymore, would it?

Comment (vgkholla) [reply ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java]: +1

File: ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java [hunk 3]
```diff
@@ -24,7 +24,7 @@
 import com.github.ambry.clustermap.MockClusterMap;
```
Comment (vgkholla) [inline ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java]: do you need any tests to check that support for BlobID V1 is not broken ? Any special cases?

File: ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java (hunk 1 above)
Comment (xiahome) [reply ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java]: Removed.

File: ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java (hunk 2 above)
Comment (xiahome) [reply ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java]: This is to test how `AmbryBlobStorageService` handle blobId string in v1.

File: ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java (hunk 3 above)
Comment (xiahome) [reply ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java]: This should be covered in `injectionAccountAndContainerForGetHeadDeleteBlobIdV1Test`

**Pull request merged**